### DISCUSSION TITLE
Skip syncing files in use

### DISCUSSION THREAD

**Discussion by ghost** [abraunegg/onedrive#2223] [Q&A] [answered]
Hello, I have this utility running as a system daemon on a box running TrueNAS Scale. Is there a way for the OneDrive client to skip syncing files that are currently in use? Our office uses Autodesk Revit design software. While a model file (*.rvt) is opened by a user, the onedrive ends up making a copy of the file with the following convention, {filename}-{hostname}.rvt and uploads to OneDrive. 
Is there a way to skip syncing of files that are currently opened by a user?

**Comment by abraunegg** [comment, marked as answer]
@VadyaRus 
> Is there a way for the OneDrive client to skip syncing files that are currently in use? Our office uses Autodesk Revit design software. While a model file (*.rvt) is opened by a user, the onedrive ends up making a copy of the file with the following convention, {filename}-{hostname}.rvt and uploads to OneDrive.

This situation is *not* caused by a local file being open.

This 'backup' (the file with the hostname in it) is created when the file on OneDrive is *different* (filesize, hash, newer timestamp) to the local file which causes a conflict. Microsoft's requirement to resolve the conflict is to rename the local file.

Now, if you are using `--resync` *at all*, this will remove the local cache state, thus, the application has no record of what is in sync, thus, additionally to preserve data and avoid potential data loss, the application creates the 'backup' of the local data to the the file with the hostname.

This situation is *not* caused by open local files that are in-use.

> Is there a way to skip syncing of files that are currently opened by a user?

You can can look at the following options:
1. Ensure that Autodesk Revit correctly is locking files that it has open locally to rule out that this is a contributing cause.
2. Run the application in verbose mode `--verbose` so you can understand the scenario by which these backup files are being created.
3. Do not use `--resync` unless the application advises you to because of configuration change. This is *very* important.
4. Use the 'config' file option `bypass_data_preservation = "true"` which will bypass the creation of these backup files - but then you are open to potential data loss scenarios where a file online will overwrite a local file with no backup .... use with caution.

**Comment by JackNWhite** [comment]
@abraunegg, what I don't understand about this problem is: if any file you've synced _from_ Sharepoint _to_ your hard drive is changed by someone else, then the remote file will always have a different size, hash and timestamp, so surely any time anyone else updates the remote copy, your local copy will be renamed and uploaded. No?

**Reply by abraunegg** [reply]
@JackNWhite 

> what I don't understand about this problem is: if any file you've synced from Sharepoint to your hard drive is changed by someone else, then the remote file will always have a different size, hash and timestamp, so surely any time anyone else updates the remote copy, your local copy will be renamed and uploaded. No?

Correct - bus specifically even even more so because of this Sharepoint 'feature' (bug) - https://github.com/OneDrive/onedrive-api-docs/issues/935 - the Sharepoint situation is amplified as this 'feature' changes the file post upload, so there is no integrity on that file regardless *who* uploaded it - so the sizes and more importantly the calculated checksum (which is the main comparison item) is *always* different .... thus, the local file will 99% of the time be renamed to preserve the local copy.
